src/tests/Makefile

```diff
@@ -1,21 +1,27 @@
 CC=gcc
-CFLAGS=-Wall -O2 -g -fno-strict-aliasing -fno-pic -fno-pie
+CFLAGS=-Wall -O2 -g -fno-strict-aliasing
 DJGPP = i586-pc-msdosdjgpp-gcc
 SOURCES = test-i386.c test-i386-code16.S test-i386-vm86.S
 ALL_SRC = $(SOURCES) $(wildcard *.h)
 
-all: test-i386 test-x86_64 test-i386.exe
+TARGETS = test-i386.exe test-i386
+LBITS := $(shell getconf LONG_BIT)
+ifeq ($(LBITS),64)
+  TARGETS += test-x86_64
```

Comment:
But how does this work with
"Only build x86_64 CPUEMU tests on 64bit"?
Clearly you build all?

Reply:
As in don't build 64 bit test on 32bit machine.

Reply:
Then you had to say "build x86_64 CPUEMU tests only on 64bit".
Otherwise it says you are going to only build
x86_64 tests, but instead you build all on 64bit.